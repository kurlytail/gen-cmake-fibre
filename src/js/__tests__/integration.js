import { execSync } from 'child_process';

describe('# integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
    });

    it('## should print help options', () => {
        let output = execSync('npm run build').toString();
        output = execSync('sgen -g `pwd`/dist/cmake-fibre.min.js -h').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design and run cmake-fibre commands', () => {
        let output = execSync('npm run build').toString();
        output = execSync(
            'sgen -g `pwd`/dist/cmake-fibre.min.js -d src/test/fixture/design.json -o testoutput'
        ).toString();
        output = output.replace(/info: Loaded generator .*cmake-fibre.min.js.*/, '');
        expect(output).toMatchSnapshot();
        output = execSync('cmake CMakeLists.txt', { cwd: 'testoutput' }).toString();
        output = execSync('make', { cwd: 'testoutput' }).toString();
    });
});
