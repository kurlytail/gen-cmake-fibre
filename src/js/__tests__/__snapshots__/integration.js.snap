// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# integration test ## should generate design 1`] = `
"info: Initializing git repository testoutput/.git
info: loading extension /Users/ashwin/pmgr/gen-cmake-fibre/node_modules/.bin/../@kurlytail/gen-cmake/scripts/../templates/extensions/cmake-minimum-required-version-default._
info: loading extension /Users/ashwin/pmgr/gen-cmake-fibre/node_modules/.bin/../@kurlytail/gen-cmake/scripts/../templates/extensions/cmake-message-versions._
info: loading extension /Users/ashwin/pmgr/gen-cmake-fibre/node_modules/.bin/../@kurlytail/gen-cmake/scripts/../templates/extensions/cmake-message-processor._
info: Generated testoutput/CMakeLists.txt from template node_modules/@kurlytail/gen-cmake/templates/CMakeLists.txt._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design and run cmake-fibre commands 1`] = `
"info: Initializing git repository testoutput/.git
info: loading extension /Users/ashwin/pmgr/gen-cmake-fibre/node_modules/.bin/../@kurlytail/gen-cmake/scripts/../templates/extensions/cmake-minimum-required-version-default._
info: loading extension /Users/ashwin/pmgr/gen-cmake-fibre/node_modules/.bin/../@kurlytail/gen-cmake/scripts/../templates/extensions/cmake-message-versions._
info: loading extension /Users/ashwin/pmgr/gen-cmake-fibre/node_modules/.bin/../@kurlytail/gen-cmake/scripts/../templates/extensions/cmake-message-processor._
info: Generated testoutput/CMakeLists.txt from template node_modules/@kurlytail/gen-cmake/templates/CMakeLists.txt._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should print help options 1`] = `
"Usage:
  node sgen [OPTION]

Options:
  -m, --map=ARG+       mapping files
  -d, --design=ARG+    design files
  -e, --extension=ARG+ extension directories or files
  -o, --output=ARG     output directory for generated files
      --overwrite=ARG  Overwrite modes, can be one of skip, regen, error                                  (default: error)
      --forceOverwrite Override overwrite flags in the generator map with the one provided on --overwrite
  -h, --help           display this help
"
`;
